vscode/package.json

```diff
@@ -497,6 +497,14 @@
         "**/bower_components": true,
         "**/*.code-search": true,
         "**/.ruby-lsp": true
+      },
+      "[ruby]": {
+        "editor.defaultFormatter": "Shopify.ruby-lsp", 
+        "editor.formatOnSave": true,
```

Comment:
Sorry, but it doesn't seem to me that this should be the default value. When saving, I want to save what I am seeing now, unless I have explicitly defined it that way. This can be problematic especially when Rubocop is very aggressive in its corrections.

Reply:
We do would want this to be `formatOnSave`, as it the [current recommendation](https://github.com/Shopify/ruby-lsp/blob/main/vscode/README.md#vs-code-configurations). We believe this is the best approach for the vast majority of developers. Saving only runs the autocorrects that RuboCop treats as safe, so it's very unlikely that this would result in incorrect code.

Reply:
Yeah, we do want this value to be `true`. The idea of the recommended defaults is for a user of the Ruby LSP to get all features out of the box, even the ones that might be turned off by default by VS Code.

For example, `formatOnSave` and `formatOnType` are both turned off by default, which leads to many people not realizing that the Ruby LSP can auto-complete `end` tokens for you as you type without the need to add any other extensions.

And remember, this is just a default recommendation. You can very easily turn this off.

Reply:
@brunoprietog if you can reproduce, then please create an issue for this so that we can address.

Reply:
It wasn't a problem with Ruby LSP, it was a problem with VS Code in general that happened to me with several file types, not just Ruby.

Reply:
We discussed this among the team and agreed to move forward with turning on as is. Our current approach with our [extension pack for Ruby](https://marketplace.visualstudio.com/items?itemName=Shopify.ruby-extensions-pack) is even more aggressive than this, offering to mutate your settings and change the values.

Turning off format on save is as simple as setting it to false for Ruby.

```json
{
  "[ruby]": {
    "editor.formatOnSave": false
  }
}
```